### PR TITLE
Some smallish translog cleanups

### PR DESCRIPTION
**Pull request by s1monw**

### CODE REVIEW

**Comment by s1monw** [comment]
@mikemccand can you take a look

File: src/main/java/org/elasticsearch/index/translog/Translog.java [hunk 1]
```diff
@@ -121,6 +121,14 @@
     private final AtomicBoolean closed = new AtomicBoolean();
     private final TranslogConfig config;
     private final String translogUUID;
+    private Callback<View> onViewClose = new Callback<View>() {
+        @Override
+        public void handle(View view) {
+            logger.trace("closing view starting at translog [{}]", view.minTranslogGeneration());
+            outstandingViews.remove(this);
```
Comment (mikemccand) [inline src/main/java/org/elasticsearch/index/translog/Translog.java]: Can we assert that remove returned true (was in fact in the set)?

File: src/main/java/org/elasticsearch/index/translog/Translog.java [hunk 2]
```diff
@@ -442,7 +450,15 @@ TranslogWriter createWriter(long fileGeneration) throws IOException {
     public Location add(Operation operation) throws TranslogException {
         ReleasableBytesStreamOutput out = new ReleasableBytesStreamOutput(bigArrays);
         try {
-            writeOperation(out, operation);
+            final BufferedChecksumStreamOutput checksumStreamOutput = new BufferedChecksumStreamOutput(out);
+            final long start = out.position();
+            out.skip(RamUsageEstimator.NUM_BYTES_INT);
+            writeOperationNoSize(checksumStreamOutput, operation);
+            long end = out.position();
+            int operationSize = (int) (out.position() - RamUsageEstimator.NUM_BYTES_INT - start);
```
Comment (mikemccand) [inline src/main/java/org/elasticsearch/index/translog/Translog.java]: You can use end instead of `out.position()` here.

**Comment by mikemccand** [comment]
LGTM

File: src/main/java/org/elasticsearch/index/translog/Translog.java (hunk 1 above)
Comment (s1monw) [inline src/main/java/org/elasticsearch/index/translog/Translog.java]: yeah good!

**Pull request merged**